src/App.tsx

```diff
@@ -1,109 +1,69 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
-
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+import { Pagination } from './components/Pagination';
 
 export const App: React.FC = () => {
+  const totalItems: number = 42;
+  const values = [3, 5, 10, 20];
+  const [numItems, setNumItems] = useState(5);
+
+  const [currentPage, setCurrentPage] = useState(1);
+
+  let from = 1;
+  let to = numItems;
+
+  if (currentPage) {
+    from = (currentPage - 1) * numItems + 1;
+    to = Math.min(currentPage * numItems, totalItems);
+  }
```

Comment:
The calculation for `from` and `to` should be encapsulated in a `useEffect` hook that depends on `currentPage` and `numItems`. This ensures that the calculation is only performed when one of these dependencies changes, rather than on every render.